### PR TITLE
Add authentication to the site

### PR DESCRIPTION
**Pull request by erichli1**
The user is shown a login/logout link depending on whether the user is logged in or logged out.

### CODE REVIEW

File: src/main/java/com/google/data/Authentication.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.google.sps.data;
```
Comment (Ortiz-Sergio) [inline src/main/java/com/google/data/Authentication.java:1]: I think we have to add the Apache License on the top? Not entirely sure.

Comment (erichli1) [reply src/main/java/com/google/data/Authentication.java:1]: @ShreeragJ Do we?

Comment (ShreeragJ) [reply src/main/java/com/google/data/Authentication.java:1]: I'm not sure, Sergio where did you see that?

File: src/main/java/com/google/sps/servlets/Login.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+
+
+@WebServlet("/login")
+public class Login extends HttpServlet {
+
+    private static final String REDIRECT_LOGIN = "/";
+    private static final String REDIRECT_LOGOUT = "/";
```
Comment (ShreeragJ) [inline src/main/java/com/google/sps/servlets/Login.java:19]: Seems weird, why are these just backslashes?

File: src/main/java/com/google/data/Authentication.java (hunk 1 above)
Comment (Ortiz-Sergio) [reply src/main/java/com/google/data/Authentication.java:1]: I don't remember exactly but I think it was a meeting about open source projects. I'm not entirely sure to be honest. Also, the java files in the example repository have the license as a comment on the top

File: src/main/java/com/google/sps/servlets/Login.java (hunk 2 above)
Comment (erichli1) [reply src/main/java/com/google/sps/servlets/Login.java:19]: I want it to just redirect to the default page which is at /, should I rename the variable to like REDIRECT_AFTER_LOGIN?

File: src/main/java/com/google/data/Authentication.java (hunk 1 above)
Comment (ShreeragJ) [reply src/main/java/com/google/data/Authentication.java:1]: Probably safe to add them in that case.

File: src/main/java/com/google/sps/servlets/Login.java (hunk 2 above)
Comment (ShreeragJ) [reply src/main/java/com/google/sps/servlets/Login.java:19]: I see, then maybe add a comment explaining that. // Redirects to default page

File: src/main/java/com/google/data/Authentication.java (hunk 1 above)
Comment (erichli1) [reply src/main/java/com/google/data/Authentication.java:1]: Sounds good, I'll add it in

File: src/main/java/com/google/sps/servlets/Login.java (hunk 2 above)
Comment (erichli1) [reply src/main/java/com/google/sps/servlets/Login.java:34]: Sounds great, just added that in!

**Pull request merged**